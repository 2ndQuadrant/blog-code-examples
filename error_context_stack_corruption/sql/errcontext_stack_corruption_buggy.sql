\set client_min_messages notice
\set VERBOSITY verbose

-- This is fine, normal return path
SELECT errcontext_stack_corruption.my_func(true, 'errcontext_buggy');

-- This returns with a bogus errcontext
SELECT errcontext_stack_corruption.my_func(false, 'errcontext_buggy');

-- So calling this unrelated code will crash when it does the errcontext
-- callback... probably? It depends hugely on stack use, where things get
-- placed, etc.
--
DO LANGUAGE plpgsql
$$
BEGIN
	RAISE EXCEPTION 'foo';
END;
$$;
