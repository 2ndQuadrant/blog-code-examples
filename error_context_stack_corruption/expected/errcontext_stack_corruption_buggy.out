\set client_min_messages notice
-- This is fine, normal return path
SELECT errcontext_stack_corruption.my_func(true, 'errcontext_buggy');
INFO:  did the thing
INFO:  after return from requested call
 my_func 
---------
 
(1 row)

-- This returns with a bogus errcontext
SELECT errcontext_stack_corruption.my_func(false, 'errcontext_buggy');
WARNING:  not doing it and leaking the error context callback pointer!
server closed the connection unexpectedly
	This probably means the server terminated abnormally
	before or while processing the request.
connection to server was lost
